codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsTxtRecord.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.dns;
+
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+public class DefaultDnsTxtRecord extends AbstractDnsRecord implements DnsTxtRecord {
+    private static final Pattern RE_KEY = Pattern.compile("`([`=\\s])");
+    private static final Pattern RE_VALUE = Pattern.compile("``");
+
+    private final String key;
+    private final String value;
+
+    public DefaultDnsTxtRecord(String name, int dnsClass, long timeToLive, String text) {
```

Comment:
How about adding a constructor that accepts `String key, String value` instead of `String text`?

Reply:
sure, let's move `decodeTxt` to `DefaultDnsRecordDecoder`, and pass the decoded `key` and `value` to `DefaultDnsTxtRecord`